app/lib/schema-data.server.ts

```diff
@@ -27,40 +28,46 @@ function isErrnoException(e: unknown): e is NodeJS.ErrnoException {
   return e instanceof Error && 'code' in e && 'errno' in e
 }
 
-export async function getVersionRefs() {
-  const releases = (await octokit.rest.repos.listReleases(repoData)).data.map(
-    (x) => ({ name: x.name, ref: x.tag_name })
-  )
-  return [...releases, { name: 'main', ref: 'main' }]
-}
-
-export async function loadJson(filePath: string, ref: string): Promise<Schema> {
-  if (!filePath) throw new Error('path must be defined')
-
-  if (extname(filePath) !== '.json')
-    throw new Response('not found', { status: 404 })
-
-  let body: Schema
-  try {
-    body = await loadContentFromGithub(filePath, ref)
-    if (body.allOf?.find((x) => x.$ref)) {
-      await loadSubSchema(body.allOf, body.$id)
-    }
-    if (body.anyOf?.find((x) => x.$ref)) {
-      await loadSubSchema(body.anyOf, body.$id)
+export const getVersionRefs = memoizee(
+  async function () {
+    const releases = (await octokit.rest.repos.listReleases(repoData)).data.map(
+      (x) => ({ name: x.name, ref: x.tag_name })
+    )
+    return [...releases, { name: 'main', ref: 'main' }]
+  },
+  { promise: true }
+)
+
+export const loadJson = memoizee(
+  async function (filePath: string, ref: string): Promise<Schema> {
+    if (!filePath) throw new Error('path must be defined')
+
+    if (extname(filePath) !== '.json')
+      throw new Response('not found', { status: 404 })
+
+    let body: Schema
+    try {
+      body = await loadContentFromGithub(filePath, ref)
+      if (body.allOf?.find((x) => x.$ref)) {
+        await loadSubSchema(body.allOf, body.$id)
+      }
+      if (body.anyOf?.find((x) => x.$ref)) {
+        await loadSubSchema(body.anyOf, body.$id)
+      }
+      if (body.oneOf?.find((x) => x.$ref, body.$id)) {
+        await loadSubSchema(body.oneOf, body.$id)
+      }
+    } catch (e) {
+      if (isErrnoException(e) && e.code === 'ENOENT') {
```

Comment:
How would the GitHub API return an OS error like this? This code is probably a remnant of when you were reading from the filesystem.

Reply:
Yup that sounds right, I'll remove that